nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -105,6 +105,9 @@ void RegulatedPurePursuitController::configure(
     node, plugin_name_ + ".max_angular_accel", rclcpp::ParameterValue(3.2));
   declare_parameter_if_not_declared(
     node, plugin_name_ + ".allow_reversing", rclcpp::ParameterValue(false));
+  declare_parameter_if_not_declared(
+    node, plugin_name_ + ".max_distance_between_iterations",
+    rclcpp::ParameterValue(10000.0));  // something really big
```

Comment:
I tried using `std::numeric_limits<double>::max` here, but for some reason it made `rclcpp::ParameterValue` think it was a `bool`? We can either make the default really big: which means that most users will not see a change in behavior with this PR, or we can make it something "sensible" that will fix intersecting paths for most users.

Reply:
@SteveMacenski what do you think of the parameter name `max_distance_between_iterations`?

Reply:
For instance, we just changed this to a parameter from the costmap size due to issues you were having, but in general, I think setting this to the costmap size is a good **maximum bound**, I wouldn't suggest any more than that, but at times, it might be worthwhile to have it less than that if you have very irregular/curvy paths or an outstandingly large local costmap.